File: src/0027-remove-element/02.go
/*************************************************************************
# File Name: 02.go
# Author: Kian Kwok
# Mail: kiankwok6@gmail.com
# Created Time: Mon Nov 16 13:18:14 2020
 ************************************************************************/

func removeElement(nums []int, val int) int {
    for i := 0; i < len(nums); {
        if nums[i] == val {  
            nums = append(nums[:i], nums[i + 1:]...)
        } else {
            i++
        }
    }
    return len(nums)
}

File: src/0191-number-of-1-bits/02.c
/*************************************************************************
# File Name: 02.c
# Author: Kian Kwok
# mail: kiankwok6@gmail.com
# Created Time: Fri Dec 20 19:27:51 2019
 ************************************************************************/

int hammingWeight(uint32_t n) {
    int bits = 0;
    while (n) {
        n = n & (n - 1);
        bits++;
    }
    return bits;
}
